### ISSUE TITLE
Containerized workflow not able to access programs in container

### ISSUE THREAD

**Issue by Ulthran** [snakemake/snakemake#2602]
**Snakemake version**

8.1.0

**Describe the bug**

When using `containerized: "docker://user/image:tag"` in a workflow with an image that has been generated using `snakemake --containerize ... > Dockerfile` the resultant container doesn't seem to have the required software installed.

**Logs**

**Minimal example**

Here's a minimal example I made:

https://github.com/Ulthran/test-containerize

with a CI run that shows the problem:

https://github.com/Ulthran/test-containerize/actions/runs/7483628651/job/20369172760

**Additional context**

Not sure if this is from some step that I'm missing or not. I've shelled into the .simg file and can't find the software that's supposed to be installed there as well.

**Comment by Ulthran** [comment]
Update, the problem is that the conda envs within the singularity images aren't being activated. If I go into the container and manually activate the environment or include the bin/ on the PATH, I can use the installed programs.

I can run `singularity exec .snakemake/singularity/fb7ef60b24791fce9cfd57c508ce6cf5.simg bash TEST.sh` with `TEST.sh` being

```
#!/bin/sh

PATH=$PATH:/conda-envs/08b0272f8c744b8bb162030774cf9917/bin/
cutadapt --version
```

and the program executes. I can also add `os.environ["PATH"] += os.pathsep + "/conda-envs/08b0272f8c744b8bb162030774cf9917/bin/"` at the start of the rule's script and it works.

Any help on this would be appreciated. This seems like something that should be internal to snakemake but isn't right now.

**Comment by Ulthran** [comment]
I found what's going wrong here: I'm using  `containerized: ...` in the workflow but then running it with `--sdm apptainer` where apparently it needs `--sdm conda apptainer` so that it knows to activate the container _then_ activate the conda env. So nothing broken but maybe a little doc update to make that clearer?

**Comment by pdimens** [comment]
I agree with @Ulthran, the documentation can be much clearer about how to specify `--sdm` when you're trying to use the `containerized: xxxx` method.
